scene/2d/physics/physics_body_2d.cpp

```diff
@@ -149,6 +149,10 @@ TypedArray<PhysicsBody2D> PhysicsBody2D::get_collision_exceptions() {
 	Array ret;
 	for (const RID &body : exceptions) {
 		ObjectID instance_id = PhysicsServer2D::get_singleton()->body_get_object_instance_id(body);
+		if (instance_id.is_null()) {
+			PhysicsServer2D::get_singleton()->body_remove_collision_exception(get_rid(), body);
+			continue;
```

Comment:
We should flag this as unlikely:

```suggestion
		if (unlikely(instance_id.is_null())) {
			PhysicsServer2D::get_singleton()->body_remove_collision_exception(get_rid(), body);
			continue;
```

We should also decide if the physics system handling this automatically is fine, or if we should recommend to the user to manually remove collision exceptions. CC @godotengine/physics. If the former, it would be nice if there was a way to prevent the error entirely, but from looking at the code I can see why that's tricky. If the latter, we should include a message like `ERR_PRINT("A previously freed physics body was used as a collision exception. The collision exception should be removed before freeing the body.")`